client/startup/i18n.js

```diff
@@ -47,6 +47,10 @@ Meteor.startup(() => {
 	});
 
 	const applyLanguage = (language = 'en') => {
+		if (language.includes('-')) {
+			language = language.replace(/-.+/, `-${ language.split('-')[1].toUpperCase() }`);
+		}
+
```

Comment:
Actually, I believe the change should be made on `filterLanguage`, more specifically [here](https://github.com/RocketChat/Rocket.Chat/pull/16757/files#diff-4096fad0d47b2cb50404023211d27bb1L24).